File: Data Structures & Algorithms/duplicate-integer/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        const hashmap = new Map();
        for (var i=0; i<nums.length; i++) {
            hashmap.set(nums[i], -1);
        }

        // check which numbers already have an assigned index.
        for (var i=0; i<nums.length; i++) {
            if (hashmap.get(nums[i]) > -1) {
                return true;
            }
            hashmap.set(nums[i], i);
        }
        return false;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        nums = set(nums)
        longest = 1
        for i in nums:
            long = 0
            if i - 1 in nums:
                continue
            if i + 1 in nums:
                j = i 
                while j in nums:
                    long += 1
                    j += 1
                if long > longest:
                    longest = long
        return longest
            
